File: DSAlgorithms/Leetcode/ts/src/arrays_hash/128-longest_consecutive_sequence.ts
const longestConsecutive = (nums: number[]): number =>{
    const set = new Set<number>();
    for(let i = 0; i < nums.length; i++)
        set.add(nums[i]);
    let longest = 0;
    set.forEach(element => {
        if(!set.has(element - 1))
        {
            let length = 1;
            while(set.has(element + length))
            ++length;
            longest = Math.max(longest, length);
        }
    })
    return longest;
};

File: DSAlgorithms/Leetcode/arrays/arrays_hash/128-longest_consecutive_sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
       std::unordered_set<int> a_set(nums.begin(), nums.end());
       int longest = 0;
       for(auto &num: a_set)
       {
           if(!a_set.count(num - 1))
           {
               int length = 1;
               while(a_set.count(num + length))
                ++length;
                longest = std::max(longest, length);
           }
       }
       return longest;
    }
};